pkg/cloudproviders/gcp/metadata.go

```diff
@@ -77,5 +78,15 @@ func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
 			},
 		},
 		Verified: verified,
+		Cluster:  clusterMetadata,
 	}, nil
 }
+
+func getClusterMetadataFromAttributes(client *metadata.Client) *storage.ClusterMetadata {
+	name, _ := client.InstanceAttributeValue("cluster-name")
+	id, _ := client.InstanceAttributeValue("cluster-uid")
+	if name != "" || id != "" {
```

Comment:
Are those really mutually exclusive or do we expect both to be set?

Reply:
GKE should set both. I guess we can treat `id` as optional, since we plan to use `name` for identification of GKE clusters.